### PR TITLE
Add Linux ROCm CI Pipeline

### PR DESCRIPTION
**Pull request by mindest**
### Description

* Add new ROCm CI pipeline (`Linux ROCm CI Pipeline`) focusing on inference.
* Resolve test errors; disable flaky tests.

based on test PR #21614.

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/linux-rocm-ci-pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+              --build_wheel \
+              --skip_submodule_sync \
+              --test --enable_onnx_tests --enable_transformers_tool_test \
+              --cmake_path /usr/bin/cmake --ctest_path /usr/bin/ctest"
+      workingDirectory: $(Build.SourcesDirectory)
+    displayName: 'Run onnxruntime unit tests'
+
+  - task: CmdLine@2
+    inputs:
+      script: |-
```
Comment (PeixuanZuo) [inline tools/ci_build/github/azure-pipelines/linux-rocm-ci-pipeline.yml:214]: Do we need to remove kernel_explorer test from rocm_training pipeline?

File: onnxruntime/test/providers/cpu/model_tests.cc [hunk 2]
```diff
@@ -537,6 +535,21 @@ ::std::vector<::std::basic_string<ORTCHAR_T>> GetParameterStrings() {
                                                 ORT_TSTR("fp16_test_tiny_yolov2"),
                                                 ORT_TSTR("fp16_test_shufflenet"),
                                                 ORT_TSTR("keras2coreml_SimpleRNN_ImageNet")};
+  static const ORTCHAR_T* rocm_disabled_tests[] = {ORT_TSTR("bvlc_alexnet"),
```
Comment (PeixuanZuo) [inline onnxruntime/test/providers/cpu/model_tests.cc:538]: Do we need to comment the reason here?

File: tools/ci_build/github/azure-pipelines/linux-rocm-ci-pipeline.yml (hunk 1 above)
Comment (mindest) [reply tools/ci_build/github/azure-pipelines/linux-rocm-ci-pipeline.yml:214]: Yes, if this new pipeline is put into use, we can then remove the dup ke test in training pipeline later.

**Comment by snnn** [comment]
It takes about 3 hours. Could it be faster? Why the code compilation takes about 2 hours?

**Comment by mindest** [comment]
> It takes about 3 hours. Could it be faster? Why the code compilation takes about 2 hours?

Compiling composable kernel takes most of the time.
I specified the arch as MI200's `gfx90a` to reduce compile time (on the orttraing-amd-ci pipeline build time is ~3 hrs.). And some tests (like kernel explorer) are copied here from the training pipeline. Should this one be enabled, I will try removing duplicates between them.

File: tools/ci_build/github/azure-pipelines/linux-rocm-ci-pipeline.yml [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+              --update \
+              --build_dir /build \
+              --build \
+              --build_shared_lib \
+              --parallel \
+              --build_wheel \
+              --enable_onnx_tests \
+              --skip_submodule_sync \
+              --use_cache \
+              --skip_tests --cmake_path /usr/bin/cmake --ctest_path /usr/bin/ctest; \
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/linux-rocm-ci-pipeline.yml:123]: Why it skip tests?

File: tools/ci_build/github/linux/docker/rocm-ci-pipeline-env.Dockerfile [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+RUN ln -sf /usr/lib/x86_64-linux-gnu/libstdc++.so.6 ${CONDA_ENVIRONMENT_PATH}/bin/../lib/libstdc++.so.6
+
+RUN pip install numpy \
+                packaging \
+                ml_dtypes==0.3.0 \
+                pytest==7.4.4 \
+                pytest-xdist \
+                pytest-rerunfailures \
+                scipy==1.10.0 \
+                numpy==1.24.1
```
Comment (snnn) [inline tools/ci_build/github/linux/docker/rocm-ci-pipeline-env.Dockerfile:86]: Numpy is listed twice here.

File: tools/ci_build/github/azure-pipelines/linux-rocm-ci-pipeline.yml (hunk 3 above)
Comment (mindest) [reply tools/ci_build/github/azure-pipelines/linux-rocm-ci-pipeline.yml:123]: We don't run tests in the build stage. They will be run and verified in test stage.

Comment (snnn) [reply tools/ci_build/github/azure-pipelines/linux-rocm-ci-pipeline.yml:123]: Then you can do "--update --build"

**Review by snnn — APPROVED** [review]

**Comment by mindest** [comment]
Thanks @snnn, @PeixuanZuo!

**Pull request merged**